### ISSUE TITLE
Failed to sign in the desired format

### ISSUE THREAD

**Issue by Imuriel310** [XML-Security/signxml#239]
Hello, so im from mexico and we need to send an xml to some governant service, i manage to resolve mi issue with an other library xmlsec, but i prefer to use signxml but i have not been able to accomplish that.  with multiple test i thnik i know where the issue is, so im signing my xml with this code 

` with open('key.pem', 'rb') as key_file:
            private_key_data = key_file.read()

        private_key = serialization.load_pem_private_key(
            private_key_data,
            password=None,  # Add your passphrase if the key is encrypted
            backend=default_backend()
        )
        with open('cer.rec.pem', 'rb') as cert_file:
            certificate = cert_file.read()
        tree = ET.parse('path_to_xml.xml')
        root = tree.getroot()

        signer = XMLSigner(
            method=methods.enveloped,
            c14n_algorithm='http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments'
        )
        # signer.sign_alg = SignatureMethod('http://www.w3.org/2000/09/xmldsig#rsa-sha1')
        signer.namespaces = {None: namespaces.ds}

        # Pass the private key object and certificate bytes to the signer
        signed_xml = signer.sign(
            root,
            key=private_key,
            # cert=certificate,
            always_add_key_value=True,
        )
        # XMLVerifier().verify(signed_xml)
        
        final_signed_xml = etree.ElementTree(signed_xml)
        final_signed_xml.write('signed_xml.xml', encoding='UTF-8', xml_declaration=False)`
this works fine but has 2 problems  or maybe 1, the most important is because inside the tag transforms 
`<Transforms>
                    <Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature" />
                    <Transform Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments" />
 </Transforms>
`
as you can see we have two items, if i send the xml with this two items the service just returns a status code 500, if i manually delete the secon transform item `<Transform Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments" />` now it can connect to the service but i dont know if this can affect the signing some hot, because i get a invalid signature error. now with xmlsec when it signs the xml it has this format 
![image](https://github.com/XML-Security/signxml/assets/78366616/d727eb79-9b65-4f7e-b563-c323f7ed4b1e)

if i use a beautify tool to get format the service will also throw 500, so idk if the problem could be the format taht signxml is giving to me.

this is the format in signxml
![image](https://github.com/XML-Security/signxml/assets/78366616/f6ba6dc2-996a-4951-a826-07c45b2f51ce)

basically after "sellado" is all in one line, i know this should not be the problem,  but eseems like the goubernament is very strict with the format,  and is just not working with signxml. so basically the answer is if:
1. there is a way to eliminate or set what is going to be inside "Transforms" tag
2.  is there a way to format the xml taht signxml is siging, i have tried with pretty print but it has the same result.

thanks a lot

>

**Comment by kislyuk** [comment]
Hello, thanks for your interest in SignXML.

You can certainly subclass the XMLSigner class to define a custom behavior that does not add the Transform node with the Algorithm. Such a behavior would be non-standard and therefore would not be accepted into the SignXML library itself - but we do provide the tools to subclass and override the method that defines this behavior.

You can't manually delete the node yourself without doing this, because deleting the node invalidates the signature. The signing process itself has to be altered before the signature is finalized.

The second issue that you describe (that the signature is all in one line) is not relevant if your counterparty is compliant. If your counterparty requires pretty-printing (which is not compliant), that can also be added by subclassing and round-tripping the XML before signing: `etree.tostring(etree.fromstring(etree.tostring(doc), pretty_print=True)), method='c14n')`.

I am going to close this issue for now since it does not appear to be a problem with SignXML. If you think otherwise, please describe the specific non-compliant behavior and we can reopen.

**Comment by Imuriel310** [comment]
thanks a lot for your response, i will try that
